plugin-server/src/main/ingestion-queues/session-recording/utils.ts

```diff
@@ -155,17 +188,48 @@ export const parseKafkaMessage = async (
 
     const headerResult = await readTokenFromHeaders(message.headers, getTeamFn)
     const token: string | undefined = headerResult.token
-    let teamIdWithConfig: null | TeamIDWithConfig = headerResult.teamIdWithConfig
+
+    if (!token) {
+        return dropMessage('no_token_in_header')
+    }
+
+    const teamIdWithConfig: null | TeamIDWithConfig = headerResult.teamIdWithConfig
 
     // NB `==` so we're comparing undefined and null
     // if token was in the headers but, we could not load team config
     // then, we can return early
-    if (!!token && (teamIdWithConfig == null || teamIdWithConfig.teamId == null)) {
+    if (teamIdWithConfig == null || teamIdWithConfig.teamId == null) {
         return dropMessage('header_token_present_team_missing_or_disabled', {
             token: token,
         })
     }
 
+    // this has to be ahead of the payload parsing in case we start dropping traffic from older versions
+    if (!!ingestionWarningProducer && !!teamIdWithConfig.teamId) {
+        const libVersion = readLibVersionFromHeaders(message.headers)
+        const minorVersion = minorVersionFrom(libVersion)
+        /**
+         * We introduced SVG mutation throttling in version 1.74.0 fix: Recording throttling for SVG-like things (#758)
+         * and improvements like jitter on retry and better batching in session recording in earlier versions
+         * So, versions older than 1.75.0 can cause ingestion pressure or incidents
+         * because they send much more information and more messages for the same recording
+         */
+        if (minorVersion && minorVersion <= 74) {
+            counterLibVersionWarning.inc()
+
+            await captureIngestionWarning(
```

Comment:
We're going to call this _alot_ Is that okay? Do we perhaps want to have a local cache of teams that we have identified with old versions? Regardless of whether Clickhouse compacts, it's potentially 100000s of kafka calls which also cost money.

Reply:
Okay with the context that this service takes care of that it is fine